### ISSUE TITLE
Crash at processEvent_SidebarWidget_

### ISSUE THREAD

**Issue by sikmir** [skyjake/lagrange#350]
Steps to reproduce:
1. Open gemini://gemini.omarpolo.com/cgi/lets-chat
2. Click on "Follow the conversation" link
3. Click on "Let's chat over Gemini!" (sidebar)

Crash details:
```
(gdb) bt
#0  0x0000000000465e69 in processEvent_SidebarWidget_ ()
#1  0x00000000004bab69 in dispatchEvent_Widget ()
#2  0x00000000004bab06 in dispatchEvent_Widget ()
#3  0x00000000004bab06 in dispatchEvent_Widget ()
#4  0x00000000004bab06 in dispatchEvent_Widget ()
#5  0x00000000004bab06 in dispatchEvent_Widget ()
#6  0x00000000004bab06 in dispatchEvent_Widget ()
#7  0x00000000004b41c7 in dispatchEvent_Window ()
#8  0x00000000004b52f9 in processEvent_Window ()
#9  0x0000000000410284 in processEvents_App ()
#10 0x00000000004114df in run_App ()
#11 0x0000000000409b63 in main ()
```

OS: Linux
Version: 1.7.1

**Comment by skyjake** [comment]
> Click on "Let's chat over Gemini!" (sidebar)

Where exactly is this "Let's chat over Gemini" in the sidebar?

**Comment by sikmir** [comment]
> Where exactly is this "Let's chat over Gemini" in the sidebar?

![scr1](https://user-images.githubusercontent.com/688044/135961547-87ab91fe-24cf-454e-a8ad-c5836839c439.jpg)

**Comment by skyjake** [comment]
The table of contents is not updated until the page load finishes, and of course a streaming page never finishes loading. I'll fix the crash, but will need to later think how/when to update the sidebar on streaming pages.

**Comment by skyjake** [comment]
To be included in v1.7.2.
